packages/dom/test/unit/lib/ElementAssertion.test.tsx

```diff
@@ -265,4 +266,36 @@ describe("[Unit] ElementAssertion.test.ts", () => {
     });
   });
 
+  describe(".toHaveFocus", () => {
+    context("when the element has focus", () => {
+      it("returns the assertion instance", () => {
+        const { getByTestId } = render(<FocusTestComponent />);
+        const input1 = getByTestId("input1");
+        input1.focus();
+        const test = new ElementAssertion(input1);
+
+        expect(test.toHaveFocus()).toBeEqual(test);
+
+        expect(() => test.not.toHaveFocus())
+          .toThrowError(AssertionError)
+          .toHaveMessage("Expected the element not to have focus.");
+      });
+    });
+
+    context("when the element does not have focus", () => {
+      it("throws an assertion error", () => {
+        const { getByTestId } = render(<FocusTestComponent />);
+        const input1 = getByTestId("input1");
+        const input2 = getByTestId("input2");
+        input1.focus();
+        const test = new ElementAssertion(input2);
```

Comment:
Just a suggestion. 
If you dont add `.focus()` , any element is focused so, I think the test could be easier like 

```suggestion
        const input1 = getByTestId("input1");
        const test = new ElementAssertion(input1);
```